src/Renderers/SpanNodeRenderer.php

```diff
@@ -42,6 +42,19 @@ public function __construct(
         $this->symfonyVersion = $symfonyVersion;
     }
 
+    public function render(): string
+    {
+        // Work around "~" being parsed as non-breaking space by rst-parser,
+        // while this is not part of the specification.
+        $spanValue = $this->span->getValue();
+        $spanValue = str_replace('~', '__TILDE__', $spanValue);
+        $this->span->setValue($spanValue);
+
+        $rendered = parent::render();
+
+        return str_replace('__TILDE__', '~', $rendered);
```

Comment:
Let's hope that we won't have content containing the actual text `__TILDE__` in the symfony documentation (or the documentation of other projects built with this builder).
Maybe we should check for `__TILDE__` in the spanValue before doing the `~` replacement and failing explicitly instead of producing a wrong output (or even smarter: selecting another non-conflicting placeholder in that case, but that might not be worth it)

Reply:
Well, I would add least add the safety check to rebuild a build failure in case the assumption is not actually safe (instead of waiting for someone to discover that some obscure page has content not rendered like the source:

```php
if (str_contains($spanValue, '__TILDE__')) {
    throw new \Exception('Cannot render content containing the text "__TILDE__" as it is used as a special placeholder in the build.');
}
```

Reply:
@stof OK, I've added your code while merging as an extra protection measure. Thanks!